recipes-connectivity/networkmanager/networkmanager-1.4.4/update-resolv-conf.patch

```diff
@@ -0,0 +1,176 @@
+#       For example: resolv.conf for interface wlan0 would be resolv.conf.wlan0
+--- a/src/dns-manager/nm-dns-manager.c
++++ b/src/dns-manager/nm-dns-manager.c
+@@ -147,6 +147,10 @@ struct _NMDnsManagerClass {
+ 	GObjectClass parent;
+ };
+ 
++#ifndef RESOLV_CONF
++#define RESOLV_CONF "/var/volatile/etc/resolv.conf"
++#endif
++
+ G_DEFINE_TYPE (NMDnsManager, nm_dns_manager, G_TYPE_OBJECT)
+ 
+ NM_DEFINE_SINGLETON_INSTANCE (NMDnsManager);
+@@ -651,6 +655,7 @@ dispatch_resolvconf (NMDnsManager *self,
+ 
+ static SpawnResult
+ update_resolv_conf (NMDnsManager *self,
++                    const char *resolv_path,
+                     char **searches,
+                     char **nameservers,
+                     char **options,
+@@ -663,7 +668,7 @@ update_resolv_conf (NMDnsManager *self,
+ 	gs_free char *content = NULL;
+ 	SpawnResult write_file_result = SR_SUCCESS;
+ 	int errsv;
+-	const char *rc_path = _PATH_RESCONF;
++	const char *rc_path = resolv_path;
+ 	nm_auto_free char *rc_path_real = NULL;
+ 
+ 	/* If we are not managing /etc/resolv.conf and it points to
+@@ -834,7 +839,7 @@ update_resolv_conf (NMDnsManager *self,
+ 		return SR_ERROR;
+ 	}
+ 
+-	if (symlink (MY_RESOLV_CONF, RESOLV_CONF_TMP) == -1) {
++	if (symlink (RESOLV_CONF, RESOLV_CONF_TMP) == -1) {
+ 		errsv = errno;
+ 		g_set_error (error,
+ 		             NM_MANAGER_ERROR,
+@@ -1161,7 +1166,7 @@ update_dns (NMDnsManager *self,
+ 		switch (priv->rc_manager) {
+ 		case NM_DNS_MANAGER_RESOLV_CONF_MAN_SYMLINK:
+ 		case NM_DNS_MANAGER_RESOLV_CONF_MAN_FILE:
+-			result = update_resolv_conf (self, searches, nameservers, options, error, priv->rc_manager);
++			result = update_resolv_conf (self, RESOLV_CONF, searches, nameservers, options, error, priv->rc_manager);
+ 			resolv_conf_updated = TRUE;
+ 			break;
+ 		case NM_DNS_MANAGER_RESOLV_CONF_MAN_RESOLVCONF:
+@@ -1178,7 +1183,7 @@ update_dns (NMDnsManager *self,
+ 		if (result == SR_NOTFOUND) {
+ 			_LOGD ("update-dns: program not available, writing to resolv.conf");
+ 			g_clear_error (error);
+-			result = update_resolv_conf (self, searches, nameservers, options, error, NM_DNS_MANAGER_RESOLV_CONF_MAN_SYMLINK);
++			result = update_resolv_conf (self, RESOLV_CONF, searches, nameservers, options, error, NM_DNS_MANAGER_RESOLV_CONF_MAN_SYMLINK);
+ 			resolv_conf_updated = TRUE;
+ 		}
+ 	}
+@@ -1186,7 +1191,7 @@ update_dns (NMDnsManager *self,
+ 	/* Unless we've already done it, update private resolv.conf in NMRUNDIR
+ 	   ignoring any errors */
+ 	if (!resolv_conf_updated)
+-		update_resolv_conf (self, searches, nameservers, options, NULL, NM_DNS_MANAGER_RESOLV_CONF_MAN_UNMANAGED);
++		update_resolv_conf (self, RESOLV_CONF, searches, nameservers, options, NULL, NM_DNS_MANAGER_RESOLV_CONF_MAN_UNMANAGED);
+ 
+ 	/* signal that resolv.conf was changed */
+ 	if (update && result == SR_SUCCESS)
+@@ -1286,6 +1291,89 @@ forget_data (NMDnsManager *self, NMDnsIP
+ }
+ 
+ static gboolean
++xc_rewrite_resolv_conf (NMDnsManager *self,
```

Comment:
The return value of this function is incorrect and triggers warnings (see further comment):
```
Jan 13 23:01:07.885387 VM Network (6): NetworkManager: <trace> [1578956467.8698] dns-mgr: update-resolv-conf: write internal file /var/volatile/etc/resolv.conf.brbridged succeeded (rc-manager=file)
Jan 13 23:01:07.885397 VM Network (6): NetworkManager: <warn>  [1578956467.8698] dns-mgr: Could not commit DNS changes: (-1) (unknown)
```